test/eoftest/eoftest_runner.cpp

```diff
@@ -0,0 +1,78 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2022 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "../utils/utils.hpp"
+#include "eoftest.hpp"
+#include <evmc/evmc.hpp>
+#include <evmone/eof.hpp>
+#include <gtest/gtest.h>
+#include <nlohmann/json.hpp>
+
+namespace json = nlohmann;
+
+namespace
+{
+struct EOFValidationTest
+{
+    struct Case
+    {
+        struct Expectation
+        {
+            evmc_revision rev;
+            bool result;
+        };
+        std::string name;
+        evmc::bytes code;
+        std::vector<Expectation> expectations;
+    };
+    std::unordered_map<std::string, Case> cases;
+};
+
+void from_json(const json::json& j, EOFValidationTest::Case& o)
+{
+    const std::optional<evmc::bytes> hex_str{evmc::from_hex(j.at("code").get<std::string>())};
+    if (!hex_str)
+        throw std::invalid_argument{"invalid hex string"};
+    const evmc::bytes& code{hex_str.value()};
+    o.code = code;
+
+    for (const auto& [rev, result] : j.at("results").items())
+    {
+        o.expectations.push_back({to_rev(rev), result.at("result").get<bool>()});
+    }
+}
+
+void from_json(const json::json& j, EOFValidationTest& o)
+{
+    if (!j.is_object() || j.empty())
+        throw std::invalid_argument{"JSON test must be an object with single key of the test name"};
+
+    const auto& j_t = *j.begin();  // Content is in a dict with the test name.
+
+    for (const auto& [name, test] : j_t.at("vectors").items())
+    {
+        o.cases.emplace(name, test.get<EOFValidationTest::Case>());
+    }
+}
+}  // namespace
+
+namespace evmone::test
```

Comment:
Wrap whole file with `evmone::test` namespace. I.e. move the anonymous namespace into the `evmone::test`.